practical-07/src/components/LoginForm.vue

```diff
@@ -0,0 +1,98 @@
+<template>
+<div class="p-4">
+  <div class="card  align-items-center">
+        <Form @submit="handleSubmit" :validation-schema="schema" class="p-6 row col-sm-4">
+            <div class="heading">Log In</div>
+ 
+            <div class="form-group mb-2">
+                <label for="email">Email address :</label>
+                <Field id="email" name="email" type="email" class="form-control" placeholder="Enter E-mail" />
+                <ErrorMessage name="email" class="text-danger" />
+            </div>
+
+            <div class="form-group mb-2">
+                <label for="password">Password :</label>
+                <Field id="password" name="password" type="password" class="form-control" placeholder="Enter Password" />
+                <ErrorMessage name="password" class="text-danger" />
+            </div>
+
+             <div class="col-auto">
+                <button type="submit" class="btn btn-success">Log In</button>
+            </div>
+            
+        </Form>
+    </div>
+</div>
+</template>
+ 
+<script>
+
+import { Form, Field, ErrorMessage } from "vee-validate";
+import * as yup from "yup";
+
+import { mapActions } from "vuex";
+export default {
+  name: 'LoginForm',
+  components: {
+  Form,
+  Field,
+  ErrorMessage,
+  },
+data() {
+    const schema = yup.object({
+    email: yup.string().email().required("Valid E-mail is Required!*"),
+    password: yup
+    .string()
+    .min(8, "Password is too short - should be 8 characters minimum")
+    .max(12, "Password is exited maximum limit - should be 8-12 characters")
+    .required()
+    });
+    return {
+      schema,
+    }
+  },
+   methods: {
+    handleSubmit(values, formActions) {
+      this.loginSubmit(values);
+      formActions.resetForm();
+    },
+    ...mapActions({
+        loginSubmit: "loginCredentials", 
+      }),
+    },
+}
+</script>
+
+<style>
+.heading{
+text-align: center;
+font-weight: bolder;
+font-size: 30px;
+}
+
+.card {
+  margin-top: 40px;
+}
+
+form {
+display: flex;
+flex-direction: column;
```

Comment:
Codacy has a fix for the issue: [Insert `··`](https://app.codacy.com/gh/mansinakrani/vuejs-practicals/pullRequest?prid=9375994)
```suggestion
  flex-direction: column;
```